### PR TITLE
Fix/additions around hostkey validation

### PR DESCRIPTION
**Pull request by rdkls**
- Fix for hostkey checking on nonstandard ssh ports
- Move server_key and fingerprint calculation into 'if hostkey_verify' block, since we only need them there
- Add ability to verify remote host against specific hostkey

### CODE REVIEW

File: ncclient/transport/ssh.py [hunk 1]
```diff
@@ -381,12 +395,32 @@ def connect(self, host, port=830, timeout=None, unknown_host_cb=default_unknown_
...
+                    key_object = key_cls(data=base64.b64decode(hostkey))
+                except paramiko.SSHException as e:
+                    # Not a key of this type - try the next
+                    pass
+            if not key_object:
+                # We've tried all known host key types and haven't found which one to use - bail
+                raise e
+            # Add the supplied hostkey to my host_keys, to be used in the following .check()
+            # key_object.get_name() is actually the key type e.g. 'ssh-rsa'
+            self._host_keys.add(host, key_object.get_name(), key_object)
```
Comment (einarnn) [inline ncclient/transport/ssh.py]: One of the checkin comments says "Add ability to verify remote host against specific hostkey", but the way the code is written means that the provided hostkey will be _added_ to any existing hostkeys? Is that the intent?

File: test/unit/test_manager.py [hunk 2]
```diff
@@ -19,6 +19,11 @@ def test_connect_ssh(self, mock_ssh):
         manager.connect(host='host')
         mock_ssh.assert_called_once_with(host='host')
 
+    @patch('ncclient.manager.connect_ssh')
```
Comment (einarnn) [inline test/unit/test_manager.py]: Could this test do more? Like verify a specific, real hostkey?

File: ncclient/transport/ssh.py (hunk 1 above)
Comment (rdkls) [reply ncclient/transport/ssh.py]: Correct - will change thanks

**Comment by rdkls** [comment]
closing in favour of new PR with different approach

**Pull request closed without merging**